### PR TITLE
[Lua] Fix shebang

### PR DESCRIPTION
**Pull request by jfcherng**
Fixes https://github.com/sublimehq/Packages/issues/2875

### CODE REVIEW

File: Lua/Lua.sublime-syntax [hunk 1]
```diff
@@ -82,6 +87,20 @@ contexts:
     - include: end
     - include: statements
 
+  shebang:
+    - match: ^\#!
+      scope: punctuation.definition.comment.lua
+      set: shebang-body
+    - match: ''
```
Comment (deathaxe) [inline Lua/Lua.sublime-syntax:94]: Immediatelly popping will cause issues once we decide to embed LUA into Markdown for instance.

grafik

That's why Bash uses

```
    - match: ^|(?=\S)
      pop: true
```

File: Lua/Lua.sublime-syntax [hunk 2]
```diff
@@ -82,6 +87,20 @@ contexts:
...
+      pop: 1
```
Comment (deathaxe) [inline Lua/Lua.sublime-syntax:95]: Not a big deal, but I'd vote to not mix pop styles a document. Either `pop: true` for all or `pop: 1`. We might switch to latter one once syntax version 2 is applied.

File: Lua/Lua.sublime-syntax [hunk 3]
```diff
@@ -82,6 +87,20 @@ contexts:
...
+
+  shebang-body:
+    - meta_scope: comment.line.shebang.lua
+    - match: \b(lua)\b
```
Comment (deathaxe) [inline Lua/Lua.sublime-syntax:99]: No capture group is needed here, so parens can be removed.

File: Lua/Lua.sublime-syntax (hunk 1 above)
Comment (jfcherng) [reply Lua/Lua.sublime-syntax:94]: Interesting knowing!

File: Lua/Lua.sublime-syntax (hunk 2 above)
Comment (jfcherng) [reply Lua/Lua.sublime-syntax:95]: There is no other Lua PR so no conflict! Let's change them to all `1`s.

**Review by deathaxe — APPROVED** [review]

**Pull request merged**